### PR TITLE
feat: removes link generation from tabs and adds aria attributes

### PR DESCRIPTION
**Pull request by codecaaron**
Update Tab component:
- Removes generated href and anchors, preventing a bug where the user could open the tab in a new browser window and 404.  LX-3668
- Adds aria-attributes and tab index's for the tab element instead of it's children, allowing the screen reader to know the correct context.
- Adds all custom classnames for the tab element states.   And allows the user to optionally fully opt out of styles.

Notes:
- Updated unit tests for new single element structure.
- This will require updates for current Tab usages as the custom styles will behave very differently now.

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.4.5-alpha.80b838.2+80b8386e<br/>@codecademy/gamut-templates@2.3.5-alpha.80b838.7+80b8386e<br/>@codecademy/gamut@2.7.2-alpha.80b838.7+80b8386e<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀Styleguide deploy preview ready!</p><p><a href=https://5e6fc9e9f8f3c867b1115dd1--codecademy-styleguide.netlify.com>https://5e6fc9e9f8f3c867b1115dd1--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e6fc9e9f8f3c867b1115dd1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e6fcd0375ff9930b239f402--codecademy-styleguide.netlify.com>https://5e6fcd0375ff9930b239f402--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e6fcd0375ff9930b239f402>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e6fd16ab6f8082a3bacd1de--codecademy-styleguide.netlify.com>https://5e6fd16ab6f8082a3bacd1de--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e6fd16ab6f8082a3bacd1de>Deploy Logs</a></p>

File: packages/gamut/src/Tabs/Tab/styles.module.scss [hunk 1]
```diff
@@ -3,45 +3,55 @@
...
-  }
-}
 .tab {
-  display: block;
+  outline: none;
+  cursor: pointer;
+}
+
+.active {
+  -webkit-font-smoothing: antialiased;
```
Comment (jakemhiller) [inline packages/gamut/src/Tabs/Tab/styles.module.scss]: not sure why this wasn't using `@include font-smoothing;` before, but it probably should

**Comment by JoshuaKGoldberg** [comment]
@codecaaron could you throw this up on a zoo server? I'd like to play around with the dashboard's tabs to see the accessibility differences

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.4.5-alpha.1401a9.8+1401a95c<br/>@codecademy/gamut-templates@2.3.5-alpha.1401a9.13+1401a95c<br/>@codecademy/gamut@2.7.2-alpha.1401a9.13+1401a95c<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e6fd92dc4535bebcdbe625a--codecademy-styleguide.netlify.com>https://5e6fd92dc4535bebcdbe625a--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e6fd92dc4535bebcdbe625a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.4.5-alpha.f4bd4b.10+f4bd4b39<br/>@codecademy/gamut-templates@2.3.5-alpha.f4bd4b.15+f4bd4b39<br/>@codecademy/gamut@2.7.2-alpha.f4bd4b.15+f4bd4b39<br/></pre>

**Comment by codecaaron** [comment]
> @codecaaron could you throw this up on a zoo server? I'd like to play around with the dashboard's tabs to see the accessibility differences

Will do!  In the meantime it should be testable in a non prod context here: https://5e6fd16ab6f8082a3bacd1de--codecademy-styleguide.netlify.com/storybook/?path=/story/component-tabs--tabs-uncontrolled

 Will update with zoo link :)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.4.5-alpha.266eee.12+266eee71<br/>@codecademy/gamut-templates@2.3.5-alpha.266eee.17+266eee71<br/>@codecademy/gamut@2.7.2-alpha.266eee.17+266eee71<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e6fdd218d7c1a0a484579bb--codecademy-styleguide.netlify.com>https://5e6fdd218d7c1a0a484579bb--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e6fdd218d7c1a0a484579bb>Deploy Logs</a></p>

**Comment by JoshuaKGoldberg** [comment]
Cool. So! This makes it quite a bit better for accessibility, but doesn't quite match the markup at https://dequeuniversity.com/library/aria/tabpanels-accordions/tabpanel2.

It's a strict set of steps forward (no steps back) but we shouldn't mark the corresponding a11y tickets as done until `aria-controls` & such are added in.

File: packages/gamut/src/Tabs/Tab/index.tsx [hunk 2]
```diff
@@ -12,52 +12,57 @@ export type TabProps = {
...
   });
   return (
-    <div className={cx(s.tabListItem, className)} role="tab">
-      <a
-        href={`${id}-panel`}
-        id={id}
-        className={tabLinkClasses}
-        onClick={e => {
-          e.preventDefault();
+    <div
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Tabs/Tab/index.tsx]: This should be a `<button>`, not a `<div>` (or a `ButtonBase` if you're into that)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-templates@2.3.5-alpha.ef2688.19+ef268841<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e78c67969480684116c0083--codecademy-styleguide.netlify.com>https://5e78c67969480684116c0083--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e78c67969480684116c0083>Deploy Logs</a></p>

File: packages/gamut/src/Tabs/Tab/index.tsx (hunk 2 above)
Comment (codecaaron) [reply packages/gamut/src/Tabs/Tab/index.tsx]: 👍

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-templates@2.3.5-alpha.c51afd.21+c51afd41<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e78c931763aba8a0d37a43d--codecademy-styleguide.netlify.com>https://5e78c931763aba8a0d37a43d--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e78c931763aba8a0d37a43d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.4.5-alpha.153a5f.17+153a5f44<br/>@codecademy/gamut-templates@2.3.5-alpha.153a5f.22+153a5f44<br/>@codecademy/gamut@2.7.2-alpha.153a5f.22+153a5f44<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e78ca8a13f286850d3cef7f--codecademy-styleguide.netlify.com>https://5e78ca8a13f286850d3cef7f--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e78ca8a13f286850d3cef7f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.4.5-alpha.786969.18+78696985<br/>@codecademy/gamut-templates@2.3.5-alpha.786969.23+78696985<br/>@codecademy/gamut@2.7.2-alpha.786969.23+78696985<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e78cc9b53202d85e5b9f17d--codecademy-styleguide.netlify.com>https://5e78cc9b53202d85e5b9f17d--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e78cc9b53202d85e5b9f17d>Deploy Logs</a></p>

**Pull request merged**